posthog/migrations/0249_add_sharingconfiguration.py

```diff
@@ -0,0 +1,89 @@
+# Generated by Django 3.2.13 on 2022-06-29 12:03
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+import posthog.models.sharing_configuration
+
+
+def create_sharing_configurations(apps, _) -> None:
+    Dashboard = apps.get_model("posthog", "Dashboard")
+    SharingConfiguration = apps.get_model("posthog", "SharingConfiguration")
+    dashboards = (
+        Dashboard.objects.filter(deprecated_is_shared=True)
+        .values("id", "team_id", "deprecated_is_shared", "deprecated_share_token")
+        .all()
+    )
+
+    batch_size = 1_000
+    sharing_configurations = [
+        SharingConfiguration(
+            team_id=dashboard["team_id"],
+            dashboard_id=dashboard["id"],
+            enabled=dashboard["deprecated_is_shared"],
+            access_token=dashboard["deprecated_share_token"],
+        )
+        for dashboard in dashboards
+    ]
+    SharingConfiguration.objects.bulk_create(sharing_configurations, batch_size=batch_size)
+
+
+def reverse(apps, schema_editor):
+    pass
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0248_add_context_for_csv_exports"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[],
+            state_operations=[
+                migrations.RenameField(model_name="dashboard", old_name="is_shared", new_name="deprecated_is_shared"),
+                migrations.AlterField(
+                    model_name="dashboard",
+                    name="deprecated_is_shared",
+                    field=models.BooleanField(db_column="is_shared", default=False),
+                ),
+                migrations.RenameField(
+                    model_name="dashboard", old_name="share_token", new_name="deprecated_share_token"
+                ),
+                migrations.AlterField(
+                    model_name="dashboard",
+                    name="deprecated_share_token",
+                    field=models.CharField(blank=True, db_column="share_token", max_length=400, null=True),
+                ),
```

Comment:
I'm pretty sure we're shooting ourselves in the foot by renaming those. We don't do that ever, and whenever we have tried renaming some columns, we have always been bitten. I get that it doesn't really rename the field in the database, and I did suggest doing exactly this for `tabs`, but even then we had problems. Like @macobo said in slack, it's normally not worth it. At all. :).

Reply:
Reverted this. Still feels confusing but happy to avoid the bigger issue if we the concern is so high. At _some point_ we will have to deal with this though otherwise the codebase will just get nastier and nastier...